File: CS50 X/problem set 6/readability/readability.py
from cs50 import get_string
import re
text=get_string("Text:")
sentences = re.split(r'[.!?]', text)
num_sentences = len(sentences) - 1
words = re.findall(r'\b\w+\b', text)
num_words = len(words)
S = num_sentences
L=sum(c.isalpha() for c in text)
W = num_words
Le=L/W*100
Se=S/W*100
print(W)
print(S)
print(L)
index = round(0.0588 * Le - 0.296 * Se - 15.8)
if index<1:
    print("Before Grade 1")
elif index>16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: CS50 X/problem set 2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text=get_string("Text: ");
    float word=1;
    float sentences=0;
    float letters=0;
    for(int i=0;i<strlen(text);i++)
    {
        if (isalpha(text[i]))
        {
            letters+=1;
        }
        else if (text[i]==' ')
        {
            word+=1;
        }
        else if (text[i]=='.'||text[i]=='?'||text[i]=='!')
        {
            sentences+=1;
        }
    }
    float L=letters/word*100;
    float S=sentences/word*100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if(index<1)
    {
        printf("Before Grade 1\n");
    }
    else if(index>16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n",index);
    }
    return 0;
}
